p2p/metrics.go

```diff
@@ -43,6 +43,19 @@ type Metrics struct {
 type metricsLabelCache struct {
 	mtx               *sync.RWMutex
 	messageLabelNames map[reflect.Type]string
```

Comment:
We could probably also pre-allocate this guy, which would allow us for getting rid of the lock.